### PR TITLE
Add validate_numbers to changeset module

### PR DESCRIPTION
**Pull request by grandynguyen**
It's back!

### CODE REVIEW

**Comment by grandynguyen** [comment]
Crap, for some reason my emacs plugin was using tabs. Will fix.

**Comment by josevalim** [comment]
Can you please squash the changes and remove the merge commit? :) Thank you!

File: lib/ecto/changeset.ex [hunk 1]
```diff
@@ -791,4 +791,75 @@ defmodule Ecto.Changeset do
...
+    float_validators = %{
+      less_than:                {&</2, :must_be_less_than},
+      greater_than:             {&>/2, :must_be_greater_than},
+      less_than_or_equal_to:    {&<=/2, :must_be_less_than_or_equal_to},
+      greater_than_or_equal_to: {&>=/2, :must_be_greater_than_or_equal_to},
+      equal_to:                 {&==/2, :must_be_equal_to},
+    }
+    integer_validators = Map.merge(float_validators, %{
+      parity:                   {&check_parity/2, :must_have_parity},
+    })
```
Comment (ericmj) [inline lib/ecto/changeset.ex]: If we move these to module attributes we dont have to `Map.merge` every time we call the function.

**Comment by grandynguyen** [comment]
Hold off on merging this. The validation errors are wrong. And I believe I just came up with a better way of implementing this.

**Comment by grandynguyen** [comment]
I'm having trouble moving the validators to module attributes. It seems as though the module attributes are evaluated before module methods? So I am unable to capture the check_parity method.

**Comment by josevalim** [comment]
@grandynguyen you can't capture the check_parity because it was not defined yet. But I was also going to ask you to drop the parity thing. :) I have never used and if someone eventually needs it, it is easy to check with a custom validation.

**Comment by josevalim** [comment]
@grandynguyen is this ready for merging? :)

**Comment by grandynguyen** [comment]
Should be ready now. See if you like using the module attribute for the validators and the handling of multiple errors.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**